### PR TITLE
logger: Port status serialization to rapidjson

### PR DESCRIPTION
**Pull request by theopolis**
This introduces a small wrapper around some `rapidjson` APIs, called `JSON` to protect us from making easy mistakes. We then port the logger's status logging serialization and deserialization to use `JSON` instead of Boost property trees.

### CODE REVIEW

File: osquery/core/json.h [hunk 1]
```diff
@@ -28,3 +37,55 @@
...
+  /// Add a string value to a JSON object.
+  void add(rapidjson::Document& line,
+           const std::string& key,
+           const std::string& value);
+
+  /// Add a size_t value to a JSON object.
+  void add(rapidjson::Document& line, const std::string& key, size_t value);
+
+  /// Add an int value to a JSON object.
+  void add(rapidjson::Document& line, const std::string& key, int value);
```
Comment (spasam) [inline osquery/core/json.h]: What about add() for bool/double value? May be there are only couple of places they might be relevant!

**Comment by osqueryer** [comment]
:-1: The commit 68355b353172d1140f5fedda0c49afea56d796cf (Job results: [803](https://jenkins.osquery.io/job/osqueryPullRequestFreeBSD/803/)) failed one or more tests (**FreeBSD**).

File: osquery/core/json.h (hunk 1 above)
Comment (theopolis) [reply osquery/core/json.h]: I'm still not married to this abstraction. :/

It is also limiting in that it only supports an array of objects. If the input document to the `add` methods was moved to the end and was an optional the default could be the managed document. Even though that works, it doesn't feel correct.

**Comment by osqueryer** [comment]
:-1: The commit 2c01c3e548eddf7107114f6dae2f8ceebf28f60c (Job results: [1001](https://jenkins.osquery.io/job/osqueryPullRequestFreeBSD/1001/)) failed one or more tests (**FreeBSD**).

**Comment by osqueryer** [comment]
:-1: The commit 2c01c3e548eddf7107114f6dae2f8ceebf28f60c (Job results: [1002](https://jenkins.osquery.io/job/osqueryPullRequestFreeBSD/1002/)) failed one or more tests (**FreeBSD**).

**Comment by osqueryer** [comment]
:no_good: The commit 2c01c3e548eddf7107114f6dae2f8ceebf28f60c (Job results: [2709](https://jenkins.osquery.io/job/osqueryPullRequestAudit/2709/)) failed the code audit or documentation test.

**Pull request merged**